packages/fxa-auth-server/lib/routes/utils/subscriptions.js

```diff
@@ -29,43 +29,131 @@ module.exports = {
     auth,
     db,
     uid,
-    client_id
+    client_id,
+    stripeHelper,
+    email
   ) {
-    const {
-      subscriptions: { productCapabilities = {}, clientCapabilities = {} } = {},
-    } = config;
-
-    const subscriptions = (await db.fetchAccountSubscriptions(uid)) || [];
-
-    const subscribedProducts = [
-      // All accounts get this product
-      PRODUCT_REGISTERED,
-      // Other products come from actual subscriptions
-      ...subscriptions.map(({ productId }) => productId),
-    ];
+    const capabilitiesToReveal = new Set();
+
+    // All accounts get this psuedo-product
+    const subscribedProducts = [PRODUCT_REGISTERED];
+
+    if (stripeHelper) {
+      await fetchSubscribedProductsFromStripe(
+        subscribedProducts,
```

Comment:
Function calls with side-effects that update passed in args feels a bit shady. Can these return the list instead? Does the one below alter it somehow too?

Reply:
Yeah, each of these subhub vs stripe "fetch" functions have a turn at adding to `subscribedProducts`. Could return a new list, but figured it was easier to just modify in place while I was refactoring things around